### DISCUSSION TITLE
Update repository info in package.jsons so Renovate can function correctly

### DISCUSSION THREAD

**Discussion by dbasilio** [TanStack/query#5818] [General]
Hello! We use [Renovate](https://docs.renovatebot.com/) to keep our dependencies up to date, and they have a built in group to keep the query monorepo up to date together. This is how they define [the group](https://docs.renovatebot.com/presets-monorepo/#monorepotanstack-query):
```json
{
  "matchSourceUrls": [
    "https://github.com/TanStack/query"
  ]
}
```
This started failing for us due to failing peerDependencies, and when I looked into it, it's because the `repository` information in the `package.json` is out of date.
```json
{
"repository": "tanstack/query",
}
``` 
This is true for the [devtools](https://github.com/TanStack/query/blob/792e0b24368b1650fdd904c9e4c85cdcf7118867/packages/react-query-devtools/package.json#L7), the [react-query package](https://github.com/TanStack/query/blob/792e0b24368b1650fdd904c9e4c85cdcf7118867/packages/react-query/package.json#L7), the [eslint package](https://github.com/TanStack/query/blob/792e0b24368b1650fdd904c9e4c85cdcf7118867/packages/eslint-plugin-query/package.json#L7) and I'm sure the others.

Based on the info from [npm](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#repository) the syntax that is being used worked when the repo was not a monorepo, but now that it is, the structure of it needs to be updated. If you look at other packages like [react](https://github.com/facebook/react/blob/main/packages/react/package.json#L32-L36) the repository section is an object that points to the URL and the package within the repository. 

I can file as an issue but this doesn't really feel like a "bug" so figured I would post it as a discussion first.

**Comment by TkDodo** [comment]
oh yeah good catch. I'd appreciate a PR to change the `repository` in all `package.json` files inside the `packages` directory towards the object syntax (as you've shown [here](https://github.com/facebook/react/blob/main/packages/react/package.json#L32-L36))

**Reply by dbasilio** [reply]
I have a branch ready but I don't have permission to push up the change @TkDodo

**Reply by TkDodo** [reply]
create a fork, push there, then make a PR from your fork to this repo please :)

**Reply by dbasilio** [reply]
https://github.com/TanStack/query/pull/5819 PR created here
